app/code/core/Mage/Eav/Model/Attribute/Data/File.php

```diff
@@ -172,6 +172,9 @@ public function validateValue($value)
         }
 
         if (!$attribute->getIsRequired() && !$toUpload) {
+            if ($toDelete) {
+                $attribute->setAttributeValidationAsPassed();
+            }
             return true;
         }
 
```

Comment:
To trace the logic, please expand to code below to show the function `compactValue`, where the delete operation is carried out.